blint/android.py

```diff
@@ -0,0 +1,277 @@
+            parent_component.properties.append(
+                Property(name="internal.appFeatures", value=features)
+            )
+        permissions = apk_permissions(app_file)
+        if permissions:
+            parent_component.properties.append(
+                Property(name="internal.appPermissions", value=permissions)
+            )
+    components = collect_files_metadata(app_file)
+    return parent_component, components
+
+
+def apk_summary(app_file):
+    """
+    Retrieve the parent component using apk summary
+    """
+    if not app_file.endswith(".apk") or not APKANALYZER_CMD:
+        return None
+    cp = exec_tool([APKANALYZER_CMD, "apk", "summary", app_file])
+    if cp and cp.returncode == 0:
+        return parse_apk_summary(cp.stdout)
+    return None
+
+
+def apk_features(app_file):
+    """
+    Retrieve the app features
+    """
+    if not app_file.endswith(".apk") or not APKANALYZER_CMD:
+        return None
+    cp = exec_tool([APKANALYZER_CMD, "apk", "features", app_file])
+    if cp and cp.returncode == 0:
+        data = cp.stdout.strip()
+        if "JAVA_TOOL_OPTIONS" in data:
+            parts = data.split("\n")
+            if parts and len(parts) > 0:
+                parts.pop(0)
+            return "\n".join(parts)
+    return None
+
+
+def apk_permissions(app_file):
+    """
+    Retrieve the app permissions
+    """
+    if not app_file.endswith(".apk") or not APKANALYZER_CMD:
+        return None
+    cp = exec_tool([APKANALYZER_CMD, "manifest", "permissions", app_file])
+    if cp and cp.returncode == 0:
+        data = cp.stdout.strip()
+        if "JAVA_TOOL_OPTIONS" in data:
+            parts = data.split("\n")
+            if parts and len(parts) > 0:
+                parts.pop(0)
+            return "\n".join(parts)
+    return None
+
+
+def collect_files_metadata(app_file):
+    """
+    Unzip the app and collect metadata
+    """
+    file_components = []
+    app_temp_dir = tempfile.mkdtemp(prefix="blint_android_app")
+    unzip_unsafe(app_file, app_temp_dir)
+    # Find and read all .version files
+    version_files = find_files(app_temp_dir, [".version"])
+    if version_files:
+        for vf in version_files:
+            file_name = os.path.basename(vf).removesuffix(".version")
+            rel_path = os.path.relpath(vf, app_temp_dir)
+            group = ""
+            name = ""
+            if "_" in file_name:
+                parts = file_name.split("_")
+                name = file_name
+                if parts and len(parts) == 2:
+                    group = parts[0]
+                    name = parts[-1]
+                else:
+                    name = name.replace("_", "-")
+                    # Patch the group name
+                    if name.startswith("kotlinx-"):
+                        group = "org.jetbrains.kotlinx"
+            with open(vf, encoding="utf-8") as fp:
+                version_data = fp.read().strip()
+                if name and version_data:
```

Comment:
@prabhu Let's close the file after we read it by deindenting starting line 162. Doesn't look like fp is used except on line 161, so no need to perform all these other operations under with.